packages/camera/camera_windows/windows/capture_controller.cpp

```diff
@@ -398,6 +399,42 @@ uint32_t CaptureControllerImpl::GetMaxPreviewHeight() const {
   }
 }
 
+enum class PlatformStreamCategory { video, photo, audio };
+
+HRESULT GetMediaSourceStreamIndex(
+    IMFCaptureSource* source, DWORD* source_stream_index,
+    PlatformStreamCategory target_stream_category) {
+  DWORD stream_count = 0;
+  HRESULT hr = source->GetDeviceStreamCount(&stream_count);
+  if (FAILED(hr)) {
+    return hr;
+  }
+
+  for (DWORD stream_index = 0; stream_index < stream_count; stream_index++) {
+    MF_CAPTURE_ENGINE_STREAM_CATEGORY stream_category;
+    hr = source->GetDeviceStreamCategory(stream_index, &stream_category);
+    if (FAILED(hr)) {
+      return hr;
+    }
+
+    if ((target_stream_category == PlatformStreamCategory::video &&
+         (stream_category == MF_CAPTURE_ENGINE_STREAM_CATEGORY_VIDEO_PREVIEW ||
+          stream_category ==
+              MF_CAPTURE_ENGINE_STREAM_CATEGORY_VIDEO_CAPTURE)) ||
```

Comment:
Why are preview and capture streams being treated interchangeably?

Reply:
That code is building a list of all relevant sources, it's not unilaterally picking a source to use. There is a huge difference between "a preview source is a potential video source" and "if the first source encountered is a preview source, always use it for video".

It doesn't seem like the approach in this PR has been sufficiently thought through; I suggest writing up a short design document explaining your planned approach so that the approach can be discussed at a high level.